### PR TITLE
Added pagination to routes

### PR DESCRIPTION
**Pull request by Nazar82**

### CODE REVIEW

File: routes/api.js [hunk 1]
```diff
@@ -7,39 +7,86 @@ const jwt = require('jsonwebtoken');
...
-    });
+    const perPage = 2;
+    const page = Number(req.query.page) || 1;
+    Recipe
+        .find({})
+        .skip((perPage * page) - perPage)
+        .limit(perPage)
+        .exec(function(err, recipes) {
+            if (err) {
+                res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
```
Comment (koldovsky) [inline routes/api.js]: should we have return here after handling error?

File: routes/api.js [hunk 2]
```diff
@@ -7,39 +7,86 @@ const jwt = require('jsonwebtoken');
...
+            }
+            Recipe.count().exec(function(err, count) {
+                if (err) {
+                    res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
+                }
```
Comment (koldovsky) [inline routes/api.js]: should we have return here after handling error?

File: routes/api.js [hunk 3]
```diff
@@ -7,39 +7,86 @@ const jwt = require('jsonwebtoken');
...
-        res.json(filteredRecipes);
-    });
+    const perPage = 2;
+    const page = Number(req.query.page) || 1;
+    Recipe.find({ main: req.query.main })
+        .skip((perPage * page) - perPage)
+        .limit(perPage)
+        .exec(function(err, recipes) {
+            if (err) {
+                res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
```
Comment (koldovsky) [inline routes/api.js]: should we have return here after handling error?

File: routes/api.js [hunk 4]
```diff
@@ -7,39 +7,86 @@ const jwt = require('jsonwebtoken');
...
+            }
+            Recipe.count({ main: req.query.main }).exec(function(err, count) {
+                if (err) {
+                    res.json({ code: HTTP_STATUS_CODES.SERVER_ERROR, error: err });
+                }
+                console.log(count);
```
Comment (koldovsky) [inline routes/api.js]: Add some descriptive info when doing console.log
Like: console.log('Page count:', count);

**Review by koldovsky — CHANGES_REQUESTED** [review]
Please review my comments

File: routes/api.js (hunk 2 above)
Comment (Nazar82) [reply routes/api.js]: Yes, to stop executing script, when the error occurs. Right?

**Pull request merged**